pages/tutorials/yaml/yaml-format.rst

```diff
@@ -0,0 +1,374 @@
+before the error was encountered. For the most part, these are internal Cantera
+functions not of direct concern here. The relevant part of this error message is
+the part between the lines of asterisks.  This message says that the YAML parser
+ran into a problem on line 4 of ``gas.yaml``. In many cases, including this one,
+the parser will fail somewhere *after* the actual problem with the input file,
+since it must continue parsing until it finds something that cannot possibly be
+valid YAML syntax. In this case, the problem from the parser's perspective is
+that the key which started on line 3 continues across a new line before it finds
+a colon that can be considered as the separator. Since a key can't be broken
+across lines like this, the parser indicates the error at the point where it
+found the colon. By looking back from the indicated point of the error, we can
+see that the problem is the missing colon in the previous line.
+
+Cantera Errors
+--------------
+
+Now let's consider the other class of errors -- ones that Cantera itself
+detects. Continuing the example above, suppose that the misspelling is
+corrected, and the input file processed again. Again an error message results,
+but this time it is from Cantera:
+
+.. code:: python
+
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+     File "/some/path/cantera/base.pyx", line 25, in cantera._cantera._SolutionBase.__cinit__
+       self._init_yaml(infile, phaseid, phases, yaml)
+     File "/some/path/cantera/base.pyx", line 49, in cantera._cantera._SolutionBase._init_yaml
+       root = AnyMapFromYamlFile(stringify(infile))
+   cantera._cantera.CanteraError:
+   ***********************************************************************
+   CanteraError thrown by Phase::addSpecies:
+   Species 'C' contains an undefined element 'C'.
+   ***********************************************************************
+
+The problem is that the phase definition specifies that all species are to be
+imported from the ``gri30`` mechanism, but only the elements H and O are
+declared. The ``gri30`` mechanism contains species composed of the elements H,
+O, C, N, and Ar. If the definition is modified to declare these additional
+elements:
+
+.. code:: yaml
+
+   phases:
+   - name: gas
+     thermo: ideal-gas
+     kinetics: gas
+     elements: [H, O, C, N, Ar]
+     species: [{gri30.yaml/species: all}]
+     reactions: [gri30.yaml/reactions]
+
+it may be imported successfully.
+
+Errors of this type do not have to be fatal, as long as you tell Cantera how you
+want to handle them. You can, for example, instruct Cantera to quietly skip
+importing any species that contain undeclared elements, instead of flagging them
+as errors. You can also specify that reactions containing undeclared species
+(also usually an error) should be skipped. This allows you to very easily
+extract a portion of a large reaction mechanism.
```

Comment:
This is covered in another place, right? It seems kind of out of place here, like dropping into some specific detail when most of the rest of the document has been about more general syntax-type stuff.

Reply:
Is there a way to illustrate this for a simpler example for this debugging illustration? I.e. something short enough so the YAML input can be displayed?

Reply:
Agreed, this is getting way ahead, into the details of the `phase` definition. I've opted to remove this paragraph.